platform/machine/qemu/cluster.go

```diff
@@ -203,6 +220,21 @@ func (qc *Cluster) NewMachine(userdata *conf.UserData) (platform.Machine, error)
 
 	plog.Debugf("NewMachine: (%s) %q", combo, qmCmd)
 
+	for i, disk := range additional.Disks {
+		additionalDiskFile, err := setupDisk(disk.Size, "")
+		if err != nil {
+			return nil, err
+		}
+		defer additionalDiskFile.Close()
+		disks = append(disks, additionalDiskFile)
+		fdnum := i + 5 // fd 4 is the primary drive
+		fdset := i + 2 // fdset 1 is used by the primary drive
+		id := fmt.Sprintf("d%d", fdnum)
+		qmCmd = append(qmCmd, "-add-fd", fmt.Sprintf("fd=%d,set=%d", fdnum, fdset),
```

Comment:
Code duplication here is not ideal.  Maybe create a closure which allocates the fd and fdset numbers and appends the args, then call it for both the primary and secondary disks.